contrib/devtools/interactive-test-runner.py

```diff
@@ -0,0 +1,237 @@
+        elif self.returncode == 0 and not self.stderr:
+            color = blue
+            status = TICK + ' Success'
+        else:
+            color = red
+            status = CROSS + ' Failed'
+
+        name = self.test.ljust(test_name_length)
+        status = status.ljust(9)
+        duration = self.finished - self.started
+
+        return color, '%s | %s | %.0f s' % (name, status, duration)
+
+
+class InteractiveTestRunner:
+
+    def __init__(self, num_jobs, passon_args, tests_order):
+        assert num_jobs >= 1
+
+        self.repo_root = self.find_git_root()
+        self.functional_test_root = os.path.join(self.repo_root, 'test', 'functional')
+        self.tests = self.read_test_runner()
+        self.num_jobs = num_jobs
+        self.passon_args = passon_args
+        self.line_length = 0
+        self.tests_order = tests_order
+        self.test_name_length = len(max(self.tests, key=len))
+
+    @staticmethod
+    def run_command(*args, **kwargs):
+        proc = subprocess.check_output(args, encoding='utf8', **kwargs)
+        return proc.stdout.strip()
+
+    def find_git_root(self):
+        return self.run_command('git', 'rev-parse', '--show-toplevel')
+
+    def find_functional_tests(self):
+        stdout = self.run_command('git', 'ls-files', '*.py', cwd=self.functional_test_root)
+        pyfiles = filter(lambda line: '/' not in line, stdout.split('\n'))
+        return list(pyfiles)
+
+    def read_test_runner(self):
+        if self.functional_test_root not in sys.path:
+            sys.path.append(self.functional_test_root)
+        from test_runner import BASE_SCRIPTS
+        return BASE_SCRIPTS
+
+    def spawn_test(self, test):
+        log_stdout = tempfile.SpooledTemporaryFile()
+        log_stderr = tempfile.SpooledTemporaryFile()
+
+        args = test.split()
+        args[0] = os.path.join(self.functional_test_root, args[0])
+        args.extend(self.passon_args)
+
+        job = subprocess.Popen(
+            args=args,
+            universal_newlines=True,
+            stdout=log_stdout,
+            stderr=log_stderr
+        )
+
+        return RunningJob(test, log_stdout, log_stderr, job)
+
+    def report_result_inline(self, test_result):
+        color, line = test_result.format(self.test_name_length)
+        missing = ' ' * max(0, self.line_length - len(line))
+        self.line_length = 0
+        print('\r%s%s' % (color(line), missing), flush=True)
+
+    def report_running(self, running_jobs):
+        def format_line(color=lambda s: s):
+            jobs = ', '.join(map(lambda j: '%s (%d)' % (color(j.test), time.time() - j.started), running_jobs))
+            return 'Running %s' % jobs
+
+        line = format_line()
+        self.line_length = max(len(line), self.line_length)
+        print('\r%s' % format_line(yellow), end='', flush=True)
+
+    def run(self, test_patterns):
+        remaining_tests = [name for name in self.tests if any([p in name for p in test_patterns])]
```

Comment:
You can use a generator for the `any` expression (notice I removed the brackets `[`, `]`):
```python
remaining_tests = [name for name in self.tests if any(p in name for p in test_patterns)]
```

Reply:
Changed in 3094b9a707c45a2250e0b7128b91000d3609e422